multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/MtaArchiveContentResolver.java

```diff
@@ -0,0 +1,137 @@
+        }
+        descriptor.setResources(list);
+    }
+
+    private boolean entryMatchesResource(Map.Entry<String, List<String>> entry, Resource resource) {
+        return entry.getValue()
+                    .stream()
+                    .anyMatch(s -> s.equals(resource.getName()));
+
+    }
+
+    private void setResourceParametersFromFile(Resource resource, Map<String, Object> fileContent) {
+        Map<String, Object> mergedProperties = mergeConfiguration(resource.getParameters(), fileContent);
+        resource.setParameters(mergedProperties);
+    }
+
+    @SuppressWarnings("unchecked")
+    private Map<String, Object> mergeConfiguration(Map<String, Object> parameters, Map<String, Object> resolvedResources) {
+        Map<String, Object> resourcesCopy = new HashMap<>(parameters);
+        resourcesCopy.putIfAbsent("config", new HashMap<String, Object>());
+
+        Map<String, Object> serviceKeyCreationParameters = (Map<String, Object>) resourcesCopy.getOrDefault("config", Collections.emptyMap());
+
+        resolvedResources.forEach(serviceKeyCreationParameters::putIfAbsent);
+        return resourcesCopy;
+    }
+
+    private void resolveRequiresDependenciesContent(String space, String appArchiveId, MtaArchiveHelper helper) {
+        Map<String, List<String>> dependencyFileAttributes = helper.getRequiresDependenciesFileAttributes();
+        for (var entry : dependencyFileAttributes.entrySet()) {
+            Map<String, Object> fileContentForEntry = fileProcessor.processFileContent(space, appArchiveId, entry.getKey());
+            mergeMtaRequiresDependencyInModules(entry, fileContentForEntry);
+        }
+    }
+
+    private void mergeMtaRequiresDependencyInModules(Map.Entry<String, List<String>> entry, Map<String, Object> fileContent) {
+        List<Module> list = new ArrayList<>();
+        for (Module module : descriptor.getModules()) {
+            resolveRequiredDependencyFileParameters(module, entry, fileContent);
+            list.add(module);
+        }
+        descriptor.setModules(list);
+    }
+
+    private void resolveRequiredDependencyFileParameters(Module module, Map.Entry<String, List<String>> entry, Map<String, Object> fileContent) {
+        for (String requiresPath : entry.getValue()) {
+            String[] splitEntry = requiresPath.split(Constants.MTA_ELEMENT_SEPARATOR);
+            String requiresDependencyModuleName = splitEntry[0];
+            String requiresDependencyName = splitEntry[1];
+            if (module.getName()
+                      .equals(requiresDependencyModuleName)) {
+                RequiredDependency requiredDependency = getRequiredDependencyForModule(module, requiresDependencyName);
+                Map<String, Object> mergedRequiredDependencyParameters = mergeConfiguration(requiredDependency.getParameters(), fileContent);
+                requiredDependency.setParameters(mergedRequiredDependencyParameters);
+                sizeTracker.incrementFileSize();
```

Comment:
here as well is good to have the check for size

Reply:
Done